hedera-node/src/main/java/com/hedera/services/store/models/EvmTransaction.java

```diff
@@ -0,0 +1,83 @@
+package com.hedera.services.store.models;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import com.google.protobuf.ByteString;
+import com.hedera.services.context.properties.GlobalDynamicProperties;
+import com.hederahashgraph.api.proto.java.AccountID;
+import com.swirlds.common.CommonUtils;
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.ethereum.core.Transaction;
+import org.hyperledger.besu.ethereum.core.Wei;
+
+import java.util.Optional;
+
+import static com.hedera.services.exceptions.ValidationUtils.validateFalse;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.CONTRACT_NEGATIVE_GAS;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.CONTRACT_NEGATIVE_VALUE;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.MAX_GAS_LIMIT_EXCEEDED;
+
+public class EvmTransaction {
+	private final GlobalDynamicProperties properties;
+	private Account sender;
+	private Account receiver;
+	private AccountID submittingNodeAccountId;
+	private long gasLimit;
+	private long amount;
+	private ByteString contractFunctionParams;
+
+	public EvmTransaction(GlobalDynamicProperties properties,
+						  Account sender,
+						  Account receiver,
+						  AccountID submittingNodeAccountId,
```

Comment:
Why do we need the submitting node account passed from the transition logic only to return it back using a getter when the transition logic asks for it?